### PR TITLE
Handler doc

### PR DESCRIPTION
**Pull request by lwgordonimos**
~~~DO NOT MERGE YET, WORK IN PROGRESS~~~

The intention here is that the documentation will be autogenerated from the code, and accessible via GitHub pages.

This should be ready to go in, bearing in mind that the documentation is a permanent work in progress.

The look and feel of the doco can be viewed in PyCharm, by browsing to the "docs/index.html" file, and selecting "Open in browser".

This is how it should be presented in GitHub Pages, which should become published at: https://aodn.github.io/python-aodncore/ when merged.

### CODE REVIEW

**Comment by anguss00** [comment]
Happy to merge this @lwgordonimos, nice one

File: aodncore/pipeline/handlerbase.py [hunk 1]
```diff
@@ -1,3 +1,178 @@
+"""
```
[the remark this answers is not in the document]
Comment (lwgordonimos) [reply aodncore/pipeline/handlerbase.py]: Agreed. I do like having all of the doco in there available when digging, but absolutely agree the important bits need to be up front and not buried.

**Comment by lwgordonimos** [comment]
@mhidas updated those few things. Would be really great to merge and test the GitHub Pages side of things. I've worked out how to add a simple table of contents/index at the top of the page, so this could easily be extended with other links (or even actual content) at a later date.

**Comment by mhidas** [comment]
@lwgordonimos Thanks. Merged!

**Pull request merged**